docs/deploying-cf-locally.md

```diff
@@ -0,0 +1,348 @@
+# How to deploy Cloud Foundry in Bosh-Lite using cf-deployment
+Deploying Cloud Foundry with Bosh-Lite is a low-cost, lightweight approach tailored for development and testing environments. Unlike AWS deployments, Bosh-Lite runs the entire Cloud Foundry stack inside a single VM on your local machine using BOSH’s local CPI (warden). This drastically reduces infrastructure requirements and speeds up prototyping. Use this setup when you need an isolated, disposable CF environment for debugging, experimentation, or learning.
+
+> 💡 Note: This environment is not production-grade.
+
+## Install VirtualBox and prerequisites
+ 
+Ensure that VirtualBox and its extension pack are installed and configured
+properly.
+
+```bash
+sudo dnf install virtualbox
+VBoxManage extpack install --replace Oracle_VirtualBox_Extension_Pack-7.1.8.vbox-extpack
+VBoxManage list extpacks
+```
+Reboot your system after installation:
+
+```bash
+sudo reboot
+```
+
+## Deploy the BOSH Director with bosh-lite 
+Follow instruction in `Bosh-Lite` [website](https://bosh.io/docs/bosh-lite/)
+
+```bash
+git clone https://github.com/cloudfoundry/bosh-deployment ~/workspace/bosh-deployment
+mkdir -p ~/deployments/vbox
+cd ~/deployments/vbox
+```
+Create the BOSH Director VM:
+
+```bash
+bosh create-env ~/workspace/bosh-deployment/bosh.yml \
+--state ./state.json \
+-o ~/workspace/bosh-deployment/virtualbox/cpi.yml \
+-o ~/workspace/bosh-deployment/virtualbox/outbound-network.yml \
+-o ~/workspace/bosh-deployment/bosh-lite.yml \
+-o ~/workspace/bosh-deployment/bosh-lite-runc.yml \
+-o ~/workspace/bosh-deployment/uaa.yml \
+-o ~/workspace/bosh-deployment/credhub.yml \
+-o ~/workspace/bosh-deployment/jumpbox-user.yml \
+--vars-store ./creds.yml \
+-v director_name=bosh-lite \
+-v internal_ip=192.168.56.6 \
+-v internal_gw=192.168.56.1 \
+-v internal_cidr=192.168.56.0/24 \
+-v outbound_network_name=NatNetwork
+```
+Check that the VMs have been created in VirtualBox:
+
+```bash
+VBoxManage list vms
+```
+
+You should see entries similar to:
+
+```bash
+"sc-7ff79fac-045c-46e2-437f-ba09d54c40bd" {ca1d6ba0-e95d-4849-9d0f-f7723386951e}
+"vm-56718394-d4c1-4b94-6ce3-3d22c712844b" {56718394-d4c1-4b94-6ce3-3d22c712844b}
+```
+
+## Configure environment variables and login
+
+```bash
+export BOSH_CLIENT=admin
+export BOSH_CLIENT_SECRET=`bosh int ./creds.yml --path /admin_password`
+bosh alias-env vbox -e 192.168.56.6 --ca-cert <(bosh int ./creds.yml --path /director_ssl/ca)
```

Comment:
does the IP need to be hardcoded in bosh-lite? or can it be anything within the local domain? I'm curious if it would make more sense to extract the IP as an env variable so that the script can be easily tailored for any environment, in case the IP is already being used, or perhaps when you might want to have 2 bosh-lite envs running at the same time... What do you think?

Reply:
Ah, I didn't know. Yeah, let's keep it this way.